### DISCUSSION TITLE
5000 basis for serial version Limit

### DISCUSSION THREAD

**Discussion by cpcpython** [pyscf/pyscf#2438] [Q&A]
Hi List
It is from manual that shows:
"unrestricted HF (~5000 basis for serial version, ~30000 basis in MPI mode)"

My question is: Can we overcome the limit of 5000 basis If I use a computer with more RAM? Does it also applicable for MPI machines  (ie. if I use a HPC with more RAM can we do calculations with more than 30000 basis sets)?

Thanks in advance
Kiran

**Comment by fishjojo** [comment]
RAM may not be the bottleneck when computing the integrals on-the-fly, but the computation itself can be expensive.
You may try the GPU code https://github.com/pyscf/gpu4pyscf to see if it works for your problem. And there is a paper about this code https://arxiv.org/pdf/2407.09700.
We are also developing an ISDF algorithm https://github.com/pyscf/pyscf-forge/pull/62, which should work for larger systems.
